sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -98,6 +98,9 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks {
         if (expected.isNaN) result.isNaN else expected == result
       case (result: Float, expected: Float) =>
         if (expected.isNaN) result.isNaN else expected == result
+      case (result: UnsafeRow, expected: GenericInternalRow) =>
```

Comment:
@mn-mikke I was just looking over compiler warnings, and noticed it claims this case is never triggered. I think it's because it would always first match the (InternalRow, InternalRow) case above. Should it go before that then?

Reply:
Hi @srowen,
```(InternalRow, InternalRow)``` case was introduced later in [21838](https://github.com/apache/spark/pull/21838) and covers the logic of the case with ```UnsafeRow```. So we can just remove the unreachable piece of code.

Reply:
Roger that, looks like Wenchen just did so. Thanks!